### ISSUE TITLE
SDC/XDC create_clock and set_false_path constraints should propigate through Yosys

### ISSUE THREAD

**Issue by litghost** [SymbiFlow/symbiflow-arch-defs#1291]
Currently SDC constraints are passed directly to VPR, which means the net names need to be the output net names from Yosys.  However, constraints should be specified on the nets in the input verilog, and then a new SDC constraints file should be output from Yosys with the new net names.

Adding this feature to the XDC plugin should work?

@tmichalak / @acomodi I believe this would allow the LiteX XDC constraints to be used throughout the flow?

@mithro FYI

**Comment by mithro** [comment]
Yes.

We can also do the clock constraint propagation through PLLs and and other boxes here.

**Comment by mithro** [comment]
@daveshah1 - FYI this is the direction we are planning on going with constraints with VPR. Do you think it would be useful for nextpnr at all?

**Comment by daveshah1** [comment]
I implemented net aliases in nextpnr, so a constraint in nextpnr can be any of the names that Yosys gives to a net. But this could be useful too

**Comment by mithro** [comment]
@daveshah1 How do you figure out the net aliases?

**Comment by daveshah1** [comment]
In the JSON, the netnames section format allows more than one name for each net. I don't know if there is a BLIF equivalent.

**Comment by litghost** [comment]
Can you close this issue once this is merged?

**Comment by tmichalak** [comment]
`create_clock` propagates though Yosys with this [SDC plugin PR](https://github.com/SymbiFlow/yosys-symbiflow-plugins/pull/27) and a [fix](https://github.com/SymbiFlow/yosys-symbiflow-plugins/pull/51).
`set_false_path` is not propagated, currently we simply write out what we get from the commands.

**Comment by tmichalak** [comment]
All related PRs have been merged. Closing.

**Comment by mithro** [comment]
Is there a separate issue so that `set_false_path` is handled correctly?
